tests/components/zha/test_device_trigger.py

```diff
@@ -180,6 +206,74 @@ async def test_if_fires_on_event(hass, mock_devices, calls):
     assert calls[0].data["message"] == "service called"
 
 
+async def test_device_offline_fires(
+    hass, zigpy_device_mock, zha_device_restored, calls
+):
+    """Test for device offline triggers firing."""
+
+    zigpy_device = zigpy_device_mock(
+        {
+            1: {
+                "in_clusters": [general.Basic.cluster_id],
+                "out_clusters": [general.OnOff.cluster_id],
+                "device_type": 0,
+            }
+        }
+    )
+
+    zha_device = await zha_device_restored(zigpy_device, last_seen=time.time())
+    await async_enable_traffic(hass, [zha_device])
+    await hass.async_block_till_done()
+
+    assert await async_setup_component(
+        hass,
+        automation.DOMAIN,
+        {
+            automation.DOMAIN: [
+                {
+                    "trigger": {
+                        "device_id": zha_device.device_id,
+                        "domain": "zha",
+                        "platform": "device",
+                        "type": "device_offline",
+                        "subtype": "device_offline",
+                    },
+                    "action": {
+                        "service": "test.automation",
+                        "data": {"message": "service called"},
+                    },
+                }
+            ]
+        },
+    )
+
+    await hass.async_block_till_done()
+    assert zha_device.available is True
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5985897)